pydantic_ai_slim/pydantic_ai/tools.py

```diff
@@ -133,6 +133,36 @@ def hitchhiker(ctx: RunContext[int], answer: str) -> str:
 Usage `ToolPrepareFunc[AgentDepsT]`.
 """
 
+ToolsPrepareFunc: TypeAlias = (
+    'Callable[[RunContext[AgentDepsT], list[ToolDefinition]], Awaitable[list[ToolDefinition] | None]]'
+)
+"""Definition of a function that can prepare the tool definition of all tools for each step.
```

Comment:
Can you please also document this feature in `docs/tools.md` where `prepare` is already documented? An example like below would be useful.

Reply:
Done! I've updated `docs/tools.md`. Let me know if you'd like any further adjustments.